### DISCUSSION TITLE
Fractional seconds on mariadb

### DISCUSSION THREAD

**Discussion by rbrigot** [theory/sqitch#655] [Q&A]
With mysql sqitch uses fractional seconds for versions 5.06.04 or higher.
But with mariadb sqitch does not use fractional seconds and deployments are very slow.

=> https://github.com/sqitchers/sqitch/blob/a63cc35717a82bce98b0151624910972658719f8/lib/App/Sqitch/Engine/mysql.pm#L220
```
has _fractional_seconds => (
    is      => 'ro',
    isa     => Bool,
    lazy    => 1,
    default => sub {
        my $dbh = shift->dbh;
        return $dbh->{mysql_serverinfo} !~ **/mariadb/i**
            && $dbh->{mysql_serverversion} >= 50604;
    },
);
```

Is it possible to add it for mariadb or force fractional seconds use ?

**Comment by rbrigot** [comment]
I would like to make a pull request but I don't know how to build the project, I don't have the Build.PL file

**Comment by theory** [comment]
Yeah it's a slight PITA to bootstrap it, but the easiest thing to do is to run

``` sh
cpanm --cpanfile dist/cpanfile --installdeps .
```

You should then have the dependencies to run, e.g.,

``` sh
export SQITCH_TEST_MYSQL_URI=db:mysql://root:password@localhost:3306/information_schema
prove -lv t/mysql.t
```

**Comment by rbrigot** [comment]
Thanks a lot for your help !

I did this pull request : https://github.com/sqitchers/sqitch/pull/658

**Reply by theory** [reply]
Awesome, thanks! Merged via #660.

**Reply by rbrigot** [reply]
Thanks ! Do you know when the next release will be created ?

**Comment by theory** [comment]
Sometime in the next few weeks, I expect.
